### ISSUE TITLE
[java] False-Positive: UnusedPrivateMethod

### ISSUE THREAD

**Issue by tmzk1005** [pmd/pmd#4107]
**Affects PMD Version:** 

6.48.0

**Description:**

A private method of inner class used by outer class, but pmd report **UnusedPrivateMethod**

**Environment**

- OS:  ubuntu 22.04

- JDK: OpenJDK11 or OpenJDK 17 

**Steps to reproduce:**

1. write demo code

```java
package demo;

public class UnusedPrivateMethodTest {

    private final InnerClass innerClass;

    public UnusedPrivateMethodTest() {
        this.innerClass = new InnerClass();
    }

    public void clear() {
        System.out.println("clear");
        doClear();
    }

    private void doClear() {
        System.out.println("do clear");
        this.innerClass.clear();
    }

    class InnerClass {

        private void clear() {
            System.out.println("inner class clear");
        }

    }


    public static void main(String[] args) {
        new UnusedPrivateMethodTest().clear();
    }

}
```

2. download pmd

```sh
cd
mkdir pmdtest
wget https://github.com/pmd/pmd/releases/download/pmd_releases%2F6.48.0/pmd-bin-6.48.0.zip
unzip pmd-bin-6.48.0.zip
```

3. execute pmd command

put demo code in `~/pmdtest/demo` and:

```sh
pmd-bin-6.48.0/bin/run.sh pmd -d demo -R rulesets/java/quickstart.xml -l java
```

console output:

```txt
8月 31, 2022 5:53:50 下午 net.sourceforge.pmd.PMD encourageToUseIncrementalAnalysis
警告: This analysis could be faster, please consider using Incremental Analysis: https://pmd.github.io/pmd-6.48.0/pmd_userdocs_incremental_analysis.html
/home/kang/pmdtest/demo/UnusedPrivateMethodTest.java:23:	UnusedPrivateMethod:	Avoid unused private methods such as 'clear()'.
```

**Comment by tmzk1005** [comment]
duplicate #4016
